include/proxsuite/proxqp/dense/utils.hpp

```diff
@@ -381,6 +395,25 @@ global_dual_residual(Results<T>& qpresults,
 
   ruiz.scale_dual_residual_in_place(
     VectorViewMut<T>{ from_eigen, qpwork.dual_residual_scaled });
+
+  ruiz.unscale_dual_in_place_eq(
+    VectorViewMut<T>{ from_eigen, qpresults.y });
+  T by = (qpmodel.b).dot(qpresults.y);
+  rhs_duality_gap = std::max(rhs_duality_gap,std::abs(by));
+  duality_gap += by;
+  ruiz.scale_dual_in_place_eq(VectorViewMut<T>{ from_eigen, qpresults.y });
+  ruiz.unscale_dual_in_place_in(
+    VectorViewMut<T>{ from_eigen, qpresults.z });
+  T zu= positive_part(qpresults.z).dot(qpmodel.u);
+  rhs_duality_gap = std::max(rhs_duality_gap,std::abs(zu));
+  duality_gap += zu;
+  T zl = negative_part(qpresults.z).dot(qpmodel.l);
+  rhs_duality_gap = std::max(rhs_duality_gap,std::abs(zl));
+  duality_gap += zl;
+  ruiz.scale_dual_in_place_in(
+    VectorViewMut<T>{ from_eigen, qpresults.z });
+  duality_gap /= sqrt_max_dim; // in order to get an a-dimensional duality gap
+  rhs_duality_gap +=1.;
```

Comment:
Why this +1?

Reply:
the associated stopping criterion is : eps_abs * rhs_duality_gap = eps_abs * (1 + max(|gx|,|xHx|,|by|,|[z]_+u|,|[z]_-l|)

Reply:
Sounds good to me. Maybe it would be better to explicitly write in the code:
`eps_abs + eps_abs * rhs_duality_gap`
for readibility

Reply:
Did you handle this remark?

Reply:
not yet, it is indeed more readable. I make the change.